### PR TITLE
Fix build and migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by urbanautomaton**
I noticed there are a number of failing tests in this project's build, and that it appears Travis CI have shut off this project at some point (😭), so there's no build feedback on PRs any more.

This PR fixes the build, and adds a GitHub Actions CI step covering all the previously-tested ruby versions. You can see the build success status for each commit [in the compare view](https://github.com/futurelearn/valid_email/compare/master...fix-ci) over in my forked repo.

Any and all feedback welcome – I did a little tidying of the build output (e.g. warnings are now only shown on CI, and rspec profiling is disabled), but if it's preferred that these are both kept by default I'm happy to add them back.

### CODE REVIEW

File: spec/email_validator_spec.rb [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
+      case error
+      when String then error
+      when Hash then error[:message]
+      else fail ArgumentError, "Unknown model error type #{error.class}"
+      end
+    end
+
+    expect(messages).to eq expected
+  end
+end
```
Comment (torrocus) [inline spec/email_validator_spec.rb:17]: I like the idea of adding a matcher :+1:

**Review by torrocus — APPROVED** [review]
LGTM

**Comment by arturoherrero** [comment]
@hallelujah Can you review this, please? It looks like this fixes https://github.com/hallelujah/valid_email/issues/120 and https://github.com/hallelujah/valid_email/issues/107 and supersedes https://github.com/hallelujah/valid_email/pull/123 and https://github.com/hallelujah/valid_email/pull/124.

**Review by hallelujah — APPROVED** [review]
Thank you for you contribution!

LGTM!

**Comment by arturoherrero** [comment]
@hallelujah @torrocus Can you push a new release to RubyGems including these changes, please?

**Comment by raghuhit** [comment]
@torrocus @hallelujah Sorry to bother you, but is there a specific date for when there is going to be the next release of the gem with these changes?

**Pull request merged**